Update the core_atmosphere cmake file to include src/framework

jim-p-w opened the pull request:
This change adds an include of the src /framework directory when building core-atmosphere.

This is required due to a dependency added in https://github.com/MPAS-Dev/MPAS-Model/pull/1359.

This fix was tested by building MPAS-Model core_atmosphere via cmake, using the gnu compiler.
This was also tested by building the mpas-bundle suite via cmake using the gnu compiler.

amstokely reviewed (changes requested):
Consider adding the following to the framework CMakeLists.txt:
```

target_include_directories(framework
    INTERFACE
        ${CMAKE_CURRENT_SOURCE_DIR}
)

```

This will automatically append the framework source directory to the core_atmosphere include path, achieving the same result without relying on relative paths.

jim-p-w commented:
> Consider adding the following to the framework CMakeLists.txt:
> target_include_directories(framework
>     INTERFACE
>         ${CMAKE_CURRENT_SOURCE_DIR}

Putting the dependency in the framework cmake file sounds much cleaner. I'll make an update.

jim-p-w commented:
The src/framework directory is now exported by the framework CMakeLists.txt, as suggested by @amstokely.

abishekg7 reviewed (approved):
Looks good to me. Checked that it builds correctly.

jim-p-w committed:
Add src/framework to list of include directories when building with CMake

In PR 1359 a new file, src/framework/mpas_halo_interface.inc is included by
several files in the src/core_atmosphere tree.
Therefore compiling core_atmosphere now requires including src/framework.

To accomplish that the CMakeLists.txt file for src/framework now exports its
source directory as an interface include directory. This adds that directory
to the compilation include path for any target which links against framework.
